### ISSUE TITLE
[Tracking] ROCM packages

### ISSUE THREAD

**Issue by Madouura** [NixOS/nixpkgs#197885]
Tracking issue for multiple new ROCm or ROCm-related packages.
Making this so I don't have to comment on every PR for every broad change.
Will update with packages as I go.

All depend on #197838

Current PRs:
- #197337
- #197342
- #197455
- #197487
- #197543
- #197693
- #197705
- #197719
- #197838

**Comment by Madouura** [comment]
If anyone is interested in helping me debug rocBLAS, here's the [current derivation](https://github.com/Madouura/nixpkgs/blob/c0cb7f58fcb69a6f062ca8cc1dcb9b0504672a01/pkgs/development/libraries/rocblas/default.nix)

**Comment by Flakebi** [comment]
Hi, thanks a lot for your work on ROCm packages!

So far, the updates where all aggregated in a single `rocm: 5.a.b -> 5.x.y` pr. I think that makes more sense than splitting the package updates into single prs for a couple of reasons:
- Often, packages have backward- (and forward-) incompatible changes, i.e. a the 5.3.0 version of rocm-runtime only works with 5.3.0 of rocm-comgr, but not with 5.2.0 or 5.4.0 (made up example).
- Nobody tests a mixture of versions, i.e. only _all_ packages at the same version are known to work.
- If I want to test hip, OpenCL and other things for an update, it’s easier to do it one time (and compile everything a single time), rather than 10 times.

tl;dr, do you mind merging all your 5.3.1 updates into a single PR?

PS: Not sure how you did the update, I usually do it with `for f in rocm-smi rocm-cmake rocm-thunk rocm-runtime rocm-opencl-runtime rocm-device-libs rocm-comgr rocclr rocminfo llvmPackages_rocm.llvm hip; nix-shell maintainers/scripts/update.nix --argstr commit true --argstr package $f; end`.

**Comment by Madouura** [comment]
I was actually afraid of the opposite being true so I split them up.
Got it, I'll aggregate them.
Thanks for the tip on the update script, that would have saved me a lot of time.

**Comment by Madouura** [comment]
Hip I think should stay separate though, since there are other changes.

**Comment by Madouura** [comment]
Done.  #198770

**Comment by Madouura** [comment]
Hold off on merging while I investigate just disabling ``BUILD_FILE_REORG_BACKWARD_COMPATIBILITY`` in this and other packages.
#198770 is okay to merge, however.

**Comment by Madouura** [comment]
#197838 is okay to merge again, see https://github.com/NixOS/nixpkgs/pull/197838#issuecomment-1301578778

**Comment by Madouura** [comment]
Updated all relevant PRs to note why we are manually defining ``CMAKE_INSTALL_<DIR>``.

**Comment by Madouura** [comment]
@acowley, @lovesegfault, and @Flakebi, could you please look at https://github.com/ROCmSoftwarePlatform/rocBLAS/issues/1277?
Pinging since you're the maintainers of ``rocm-llvm``.
I think we may need to adjust ``rocm-llvm`` in some way.

**Comment by Madouura** [comment]
@lovesegfault Mass change to ``rocmVersion`` done.
Naturally, all the relevant PRs are broken until #198770 is merged, if testing please make sure to cherry-pick or rebase on that PR.

**Comment by Madouura** [comment]
Rolled everything into #198770

**Comment by Madouura** [comment]
May want to change ``rec`` to ``finalAttrs``, mainly since we do a cross-derivation check to determine if the derivation is broken.

**Comment by Madouura** [comment]
@Flakebi Once #202373 is merged, would you mind standardizing the older rocm derivations to the new ones?
Assuming @lovesegfault and @acowley are okay with it, of course.

**Comment by acowley** [comment]
Of course!

**Comment by Madouura** [comment]
rocRAND already includes hipRAND and hipRAND has no releases or tags so until further notice, no separate hipRAND.

**Comment by Madouura** [comment]
If anyone is interested in helping me debug the ``roctracer`` derivation, see: https://github.com/Madouura/nixpkgs/blob/05d79d8d63c96f06cf53a00e6e83ab2354fe2bce/pkgs/development/libraries/roctracer/default.nix

**Comment by Madouura** [comment]
Last few packages with a ROCm version are gonna be put into a PR soon.
After that, going to work on implementing any missed documention, ``impureTests``, and ``-DAMDGPU_TARGETS``.
Then, after that, one final possible rocm-related (not anything like ``lit`` this time) rebuild with cleanup/optimization/patchelf fixes for all the rocm packages with everything I've learned.

**Comment by Madouura** [comment]
If anyone is interested in helping me add rocm support to tensorflow, here is the current WIP: https://github.com/Madouura/nixpkgs/tree/dev/rocm

**Comment by LunNova** [comment]
Don't have time to help, it's really cool that you're working on this and I'll probably benefit from it at some point in the future if it gets in! Thanks for putting in all the effort here.

**Comment by Madouura** [comment]
Should we make a ``rocmPackages`` attrset similar to ``cudaPackages``?
I'm already looking at mass changes with using ``strictDeps`` and it would be nice to include ``llvmPackages_rocm.<derivation>`` in it since the llvm package for rocm is completely custom.
It might also be nice since we could separate major changes (i.e.: 5.x.x -> 6.x.x) and keep both.

**Comment by Flakebi** [comment]
A package set for rocm sounds good to me.
Thanks a lot for all of your work on ROCm packages!

**Comment by happysalada** [comment]
@Madouura thanks a lot for your work on this!
I was looking at testing rocm support for torch with openai-whisper.
I've added the following package
```
          (openai-whisper.overrideAttrs (finalAttrs: previousAttrs: {
            torch = python3Packages.torchWithRocm;
          }))
```

however when I try to run it ` whisper cougar_town.mp3 --model large-v2 --device 'cuda'`
then I get
```
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
```

when I type rocminfo, It does show me a gpu though.
If there is anything that can be tried, I'd be happy to test stuff.

**Comment by Madouura** [comment]
> @Madouura thanks a lot for your work on this! I was looking at testing rocm support for torch with openai-whisper. I've added the following package
> 
> ```
>           (openai-whisper.overrideAttrs (finalAttrs: previousAttrs: {
>             torch = python3Packages.torchWithRocm;
>           }))
> ```
> 
> however when I try to run it ` whisper cougar_town.mp3 --model large-v2 --device 'cuda'` then I get
> 
> ```
> RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
> ```
> 
> when I type rocminfo, It does show me a gpu though. If there is anything that can be tried, I'd be happy to test stuff.
> 
> (using cuda as device is coming from the guide [here](https://pytorch.org/docs/stable/notes/hip.html)

Strange, I can't find any reason this would be happening in the code.
Try running this with torchWithRocm and paste the output so I can make sure it's not on your side.
``benchmark.py``
```python
import torch, timeit

print(f"CUDA support: {torch.cuda.is_available()} (Should be \"True\")")
print(f"CUDA version: {torch.version.cuda} (Should be \"None\")")
print(f"HIP version: {torch.version.hip} (Should contain \"5.4\")")

# Storing ID of current CUDA device
cuda_id = torch.cuda.current_device()
print(f"Current CUDA device ID: {torch.cuda.current_device()}")
print(f"Current CUDA device name: {torch.cuda.get_device_name(cuda_id)} (Should be AMD, not NVIDIA)")

def batched_dot_mul_sum(a, b):
    '''Computes batched dot by multiplying and summing'''
    return a.mul(b).sum(-1)


def batched_dot_bmm(a, b):
    '''Computes batched dot by reducing to bmm'''
    a = a.reshape(-1, 1, a.shape[-1])
    b = b.reshape(-1, b.shape[-1], 1)
    return torch.bmm(a, b).flatten(-3)

x = torch.randn(10000, 1024, device='cuda')

t0 = timeit.Timer(
    stmt='batched_dot_mul_sum(x, x)',
    setup='from __main__ import batched_dot_mul_sum',
    globals={'x': x})

t1 = timeit.Timer(
    stmt='batched_dot_bmm(x, x)',
    setup='from __main__ import batched_dot_bmm',
    globals={'x': x})

# Ran each twice to show difference before/after warmup
print(f'mul_sum(x, x):  {t0.timeit(100) / 100 * 1e6:>5.1f} us')
print(f'mul_sum(x, x):  {t0.timeit(100) / 100 * 1e6:>5.1f} us')
print(f'bmm(x, x):      {t1.timeit(100) / 100 * 1e6:>5.1f} us')
print(f'bmm(x, x):      {t1.timeit(100) / 100 * 1e6:>5.1f} us')
```

**Comment by Madouura** [comment]
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/transformers/default.nix
This may also need to be changed to ``torchWithRocm``.

**Comment by happysalada** [comment]
Thanks a lot for the help, I got rid of the runtime error with your help.
I needed to use override and not override attrs, so for anyone else I needed
```
          (let torchWithRocm = python3Packages.torchWithRocm;
            in
          openai-whisper.override {
            torch = torchWithRocm;
            transformers = python3Packages.transformers.override {
              torch = torchWithRocm;
            };
          })
```

However, now when I run the program, It seems to get stuck.

I was looking at the gputargets inside hip, and indeed mine is missing (I've got gfx90c)
Do you think it would be okay to add it ?
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/compilers/hip/default.nix#L166
The comment looks old, and looking at the supported GPUs, all the gfx9 should be included https://community.amd.com/t5/knowledge-base/amd-rocm-hardware-and-software-support-document/ta-p/489937
Is there anything to watch out for ?
Let me know if you think I've missed something.

**Comment by luizirber** [comment]
This is amazing, thanks for all the hard work @Madouura! I've been watching this issue for a long time because I never managed to use ROCm with my (unsupported?) RX 5500 XT, but I can at least start the benchmark now =]

It does fail with missing `TensileLibrary.dat` in rocBLAS, tho:
```
$ python benchmark.py
CUDA support: True (Should be "True")
CUDA version: None (Should be "None")
HIP version: 5.4.22803-0 (Should contain "5.4")
Current CUDA device ID: 0
Current CUDA device name: AMD Radeon RX 5500 XT (Should be AMD, not NVIDIA)
mul_sum(x, x):   84.2 us
mul_sum(x, x):   10.9 us

rocBLAS error: Cannot read /nix/store/923xk8qd08k8vyyy98zn91p0dw627xcq-rocblas-5.4.2/lib/rocblas/library/TensileLibrary.dat: No such file or directory
Aborted (core dumped)
```

I used this `flake.nix` to create a shell:
```
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
  };

  outputs = { self, nixpkgs }:
    let pkgs = nixpkgs.legacyPackages."x86_64-linux";
    in {
      devShells.x86_64-linux.default = pkgs.mkShell {
        buildInputs = [
          pkgs.python3Packages.torchWithRocm
          pkgs.rocm-smi
          pkgs.nvtop-amd
        ];
      };
    };
}
```

How do I get the `.dat` files?

Thanks again!

**Comment by Madouura** [comment]
> Thanks a lot for the help, I got rid of the runtime error with your help. I needed to use override and not override attrs, so for anyone else I needed
> 
> ```
>           (let torchWithRocm = python3Packages.torchWithRocm;
>             in
>           openai-whisper.override {
>             torch = torchWithRocm;
>             transformers = python3Packages.transformers.override {
>               torch = torchWithRocm;
>             };
>           })
> ```
> 
> However, now when I run the program, It seems to get stuck.
> 
> I was looking at the gputargets inside hip, and indeed mine is missing (I've got gfx90c) Do you think it would be okay to add it ? https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/compilers/hip/default.nix#L166 The comment looks old, and looking at the supported GPUs, all the gfx9 should be included https://community.amd.com/t5/knowledge-base/amd-rocm-hardware-and-software-support-document/ta-p/489937 Is there anything to watch out for ? Let me know if you think I've missed something.

``gfx90c`` likely isn't supported, but go ahead and try adding it to the ``gpuTargets`` and see if it works.

> This is amazing, thanks for all the hard work @Madouura! I've been watching this issue for a long time because I never managed to use ROCm with my (unsupported?) RX 5500 XT, but I can at least start the benchmark now =]
> 
> It does fail with missing `TensileLibrary.dat` in rocBLAS, tho:
> 
> ```
> $ python benchmark.py
> CUDA support: True (Should be "True")
> CUDA version: None (Should be "None")
> HIP version: 5.4.22803-0 (Should contain "5.4")
> Current CUDA device ID: 0
> Current CUDA device name: AMD Radeon RX 5500 XT (Should be AMD, not NVIDIA)
> mul_sum(x, x):   84.2 us
> mul_sum(x, x):   10.9 us
> 
> rocBLAS error: Cannot read /nix/store/923xk8qd08k8vyyy98zn91p0dw627xcq-rocblas-5.4.2/lib/rocblas/library/TensileLibrary.dat: No such file or directory
> Aborted (core dumped)
> ```
> 
> I used this `flake.nix` to create a shell:
> 
> ```
> {
>   inputs = {
>     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
>   };
> 
>   outputs = { self, nixpkgs }:
>     let pkgs = nixpkgs.legacyPackages."x86_64-linux";
>     in {
>       devShells.x86_64-linux.default = pkgs.mkShell {
>         buildInputs = [
>           pkgs.python3Packages.torchWithRocm
>           pkgs.rocm-smi
>           pkgs.nvtop-amd
>         ];
>       };
>     };
> }
> ```
> 
> How do I get the `.dat` files?
> 
> Thanks again!

IIRC the RX 5500 XT isn't supported on ``rocBLAS``, you're out of luck until it is.
Make an issue on the ``rocBLAS`` and see if they can help. Sorry!

**Comment by Madouura** [comment]
See https://github.com/RadeonOpenCompute/ROCm/blob/77cbac4abab13046ee93d8b5bf410684caf91145/README.md#library-target-matrix for what is definitely supported.

**Comment by Madouura** [comment]
Okay nevermind, 5500 is supported on rocBLAS.
I'll look into it, still make that issue if you can please.

**Comment by Madouura** [comment]
@luizirber Did you build rocBLAS yourself or did you fetch from the nixos cache?
The tensile stuff should be generated regardless and IIRC is working on hydra, so this shouldn't be happening.

**Comment by Madouura** [comment]
@happysalada This may be of some use to you.
https://github.com/RadeonOpenCompute/ROCm/issues/1743

**Comment by Madouura** [comment]
@luizirber Try setting these both to false: https://github.com/NixOS/nixpkgs/blob/6e30230a696728c3b14a477911293e257de4873d/pkgs/development/libraries/rocblas/default.nix#L22-L23
This still shouldn't be happening on the benchmark, but whatever it is on your side this _should_ fix it.

**Comment by Madouura** [comment]
> This is amazing, thanks for all the hard work @Madouura! I've been watching this issue for a long time because I never managed to use ROCm with my (unsupported?) RX 5500 XT, but I can at least start the benchmark now =]
> 
> It does fail with missing `TensileLibrary.dat` in rocBLAS, tho:
> 
> ```
> $ python benchmark.py
> CUDA support: True (Should be "True")
> CUDA version: None (Should be "None")
> HIP version: 5.4.22803-0 (Should contain "5.4")
> Current CUDA device ID: 0
> Current CUDA device name: AMD Radeon RX 5500 XT (Should be AMD, not NVIDIA)
> mul_sum(x, x):   84.2 us
> mul_sum(x, x):   10.9 us
> 
> rocBLAS error: Cannot read /nix/store/923xk8qd08k8vyyy98zn91p0dw627xcq-rocblas-5.4.2/lib/rocblas/library/TensileLibrary.dat: No such file or directory
> Aborted (core dumped)
> ```
> 
> I used this `flake.nix` to create a shell:
> 
> ```
> {
>   inputs = {
>     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
>   };
> 
>   outputs = { self, nixpkgs }:
>     let pkgs = nixpkgs.legacyPackages."x86_64-linux";
>     in {
>       devShells.x86_64-linux.default = pkgs.mkShell {
>         buildInputs = [
>           pkgs.python3Packages.torchWithRocm
>           pkgs.rocm-smi
>           pkgs.nvtop-amd
>         ];
>       };
>     };
> }
> ```
> 
> How do I get the `.dat` files?
> 
> Thanks again!

I should mention I did exactly that setup just now and am still not getting any rocBLAS errors.
I wonder if it's your setup or there's some switch toggled if the GPU is different?

**Comment by happysalada** [comment]
Hey thanks a lot !

I tried what they recommended on the issue and it didnt work. They also mentioned that likely my gpu wouldnt be supported. Ill keep an eye out for when it is.
Thanks for your great work on this !

**Comment by CyberShadow** [comment]
> If anyone is interested in helping me add rocm support to tensorflow, here is the current WIP: [Madouura@8f667a0](https://github.com/Madouura/nixpkgs/commit/8f667a0df2a7c29a8fb4323d1524b3314cbd9068)

Hi!

- This commit doesn't seem to be on any branch, is that right? Somehow I managed to clone it anyway.
- I see this uses [tensorflow/tensorflow](https://github.com/tensorflow/tensorflow/) and not [ROCmSoftwarePlatform/tensorflow-upstream](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream). Is that right? I heard AMD was upstreaming ROCm support into TF but looking at I see it's still active and there are many commits that are not in the upstream repo.

**Comment by mustafasegf** [comment]
hi there, rocm 5.6 released couple days ago. Can we upgrade to 5.6 or at least 5.5 to suport rdna 3 graphics card?

**Comment by Madouura** [comment]
> hi there, rocm 5.6 released couple days ago. Can we upgrade to 5.6 or at least 5.5 to suport rdna 3 graphics card?

There's some issues upgrading that I don't have time for at the moment, but here's a patch to get anyone who needs it fixed now started.
[rocm.patch.txt](https://github.com/NixOS/nixpkgs/files/11945635/rocm.patch.txt)

**Comment by mustafasegf** [comment]
> > hi there, rocm 5.6 released couple days ago. Can we upgrade to 5.6 or at least 5.5 to suport rdna 3 graphics card?
> 
> There's some issues upgrading that I don't have time for at the moment, but here's a patch to get anyone who needs it fixed now started. [rocm.patch.txt](https://github.com/NixOS/nixpkgs/files/11945635/rocm.patch.txt)

Thank you very much for the patch and for the work you put in rocm suopport! I hope you can fix the issue soon regarding the upgrade!

**Comment by Madouura** [comment]
> > If anyone is interested in helping me add rocm support to tensorflow, here is the current WIP: [Madouura@8f667a0](https://github.com/Madouura/nixpkgs/commit/8f667a0df2a7c29a8fb4323d1524b3314cbd9068)
> 
> Hi!
> 
>     * This commit doesn't seem to be on any branch, is that right? Somehow I managed to clone it anyway.
> 
>     * I see this uses [tensorflow/tensorflow](https://github.com/tensorflow/tensorflow/) and not [ROCmSoftwarePlatform/tensorflow-upstream](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream). Is that right? I heard AMD was upstreaming ROCm support into TF, but looking at it I see it's still active and there are many commits that are not in the upstream repo.

It would be preferable to use the standard `tensorflow`. That said if it doesn't work even after the update (it's been half a year) I'm working on, I think I'll make a variant in `rocmPackages`.

**Comment by kurnevsky** [comment]
Hi. Thanks for maintaining rocm for nix!

When I try to use `torchWithRocm` I got the following error:

```
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' naive_conv.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: naive_conv.cpp
MIOpen(HIP): Warning [BuildHip] hip runtime failed to load.
Error: Please provide architecture for which code is to be generated.
MIOpen Error: /build/source/src/hipoc/hipoc_program.cpp:304: Code object build failed. Source: naive_conv.cpp
```

Any idea what should be in the environment? I tried adding recent `meta.rocm-all` but it didn't help.

**Comment by Madouura** [comment]
Looks like it's a nix derivation.
Could you give me it so I can replicate this?

**Comment by kurnevsky** [comment]
> Looks like it's a nix derivation.
> Could you give me it so I can replicate this?

It's just my [nix configs](https://github.com/kurnevsky/nixfiles) where I added `torchWithRocm` to the environment packages.

> From a cursory look, either hip is actually not being found or the GPU target is not being specified.

I tried to specify target like [this](https://github.com/kurnevsky/nixfiles/blob/696cb791984b4ce21831dba60391f7e63ff0fb71/machines/pc/configuration.nix#L88-L97) but as I understand this target is enabled by default now. Anyway the same result with and without it.

> Also please list your GPU(s).

It's `7900 XTX`.

**Comment by Madouura** [comment]
Unless I'm seeing this wrong or not looking enough, at https://github.com/kurnevsky/nixfiles/blob/bcbd6f98d40bb0bd2e11fb7aae4ff547f01b8f26/modules/desktop.nix#L259C13-L259C13:
You're not using torchWithRocm, just pytorch.
Try this: https://github.com/NixOS/nixpkgs/issues/197885#issuecomment-1419995566

**Comment by kurnevsky** [comment]
Yeah, I just didn't commit the change to `torchWithRocm` since it doesn't work :)

> Try this: https://github.com/NixOS/nixpkgs/issues/197885#issuecomment-1419995566

It works fine:

```
CUDA support: True (Should be "True")
CUDA version: None (Should be "None")
HIP version: 5.7.31921- (Should contain "5.7")
Current CUDA device ID: 0
Current CUDA device name: AMD Radeon RX 7900 XTX (Should be AMD, not NVIDIA)
mul_sum(x, x):   92.7 us
mul_sum(x, x):    5.2 us
bmm(x, x):      430.9 us
bmm(x, x):        8.7 us
```

But I still have the error when trying to use diffusers or even when trying to run my [custom network](https://github.com/pointsgame/oppai-rs/blob/master/zero-torch/oppai_net.py).

**Comment by Madouura** [comment]
I'm not entirely sure how I can help.
Is this an issue with whatever you're trying to use ``torchWithRocm`` with or with ``torchWithRocm`` itself?

**Comment by kurnevsky** [comment]
With `torchWithRocm` itself. Could you try running this script:

```python
import torch as trc
import os
gpu = trc.device("cuda")

ex1 = trc.zeros(1, 1, 5, 5)
ex1[0, 0, :, 2] = 1

conv1 = trc.nn.Conv2d(in_channels=1, out_channels=1, kernel_size=3)
res1 = conv1(ex1)
print(res1)  #shows result

ex2 = ex1.to(gpu)
conv2 = conv1.to(gpu)

res2 = conv2(ex2) #error here
print(res2)
```

Does it work for you?

**Comment by kurnevsky** [comment]
> I believe your GPU is GFX11?

**Comment by Madouura** [comment]
New tensorflow-rocm WIP at https://github.com/Madouura/nixpkgs/commit/344aa780809455f545eae4895bed72e3e9af0de6.
Current blocking factor is an LLVM mismatch.

**Comment by Madouura** [comment]
@Flakebi I have some preliminary ``impureTests`` stuff at https://github.com/Madouura/nixpkgs/blob/5f0c4239766cc2e6968b7aec6a881f367be7c86d/pkgs/development/rocm-modules/5/rocm-thunk/generic.nix#L60-L64 as well as some other stuff.
Tell me if you think this is the best way to go forward please.

**Comment by Madouura** [comment]
I'm going to take a bit of a break from ROCm and work on another project.
I'll try to work on the major updates/upgrades here and there, but until early-mid next year the other project is going to be my focus.
If there's any major issues or if you just need something explained, don't hesitate to ping me.

**Comment by gjz010** [comment]
> Hi. Thanks for maintaining rocm for nix!
> 
> When I try to use `torchWithRocm` I got the following error:
> 
> ```
> MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' naive_conv.cpp: HIPRTC_ERROR_COMPILATION (6)
> MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: naive_conv.cpp
> MIOpen(HIP): Warning [BuildHip] hip runtime failed to load.
> Error: Please provide architecture for which code is to be generated.
> MIOpen Error: /build/source/src/hipoc/hipoc_program.cpp:304: Code object build failed. Source: naive_conv.cpp
> ```
> 
> Any idea what should be in the environment? I tried adding recent `meta.rocm-all` but it didn't help.

Same problem here with same GPU (7900 XTX). After some `strace` on your minimal example I noticed that:
```
openat(AT_FDCWD, "/nix/store/mkih90ygzxczv4k0fn6gapgi7i7wy292-rocm-llvm-libunwind-5.7.1/lib/libamdhip64.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
...
openat(AT_FDCWD, "./libamdhip64.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./libamdhip64.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "MIOpen(HIP): Error [Compile] 'hi"..., 145MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' naive_conv.cpp: HIPRTC_ERROR_COMPILATION (6)
) = 145
```

It appears that some important `libamdhip64.so` is not added to runtime library path:
```bash
ls $(dirname $(nix-shell -p rocmPackages.meta.rocm-hip-runtime --run "which hipcc"))/../lib/libamdhip64.so
# /nix/store/09ic1qizx0aacml0vi83k9lgq23fz0wg-rocm-hip-runtime-meta/bin/../lib/libamdhip64.so
```


By setting environment variable manually:
```bash
export LD_LIBRARY_PATH=/nix/store/bz15zrilgr04ghdiz4cd73sam5wvmhhw-clr-5.7.1/lib/
```
The problem is temporarily fixed and I can now run Stable Diffusion WebUI.

**Comment by Madouura** [comment]
[ROCm 6.0.0](https://github.com/ROCm/ROCm/releases/tag/rocm-6.0.0) has been released.
``rocmPackages_5`` is now in maintenance-mode. I will eventually backport the changes I am making with ``rocmPackages_6`` to ``rocmPackages_5``, however it is not a high priority.

**Comment by kurnevsky** [comment]
> By setting environment variable manually

Interesting - now pytorch works for me, but it doesn't seem to work correctly. I'm trying to generate an image from sdxl+lora with diffusers, and it generates an incorrect image...

I tried identical code and model with manually defined seeds in google colab with cuda - it works there. Also seems to work locally on cpu with f32 types.

**Comment by sersorrel** [comment]
The `export LD_LIBRARY_PATH=/nix/store/...-clr-5.7.1/lib` solution fixed the same [`torchWithRocm` problem](https://github.com/NixOS/nixpkgs/issues/197885#issuecomment-1774700469) for me, also with a 7900 XTX. I couldn't see how you got that path – it's returned by `nix build --print-out-paths nixpkgs#rocmPackages.clr`, right?

**Comment by ScatteredRay** [comment]
Hey, giving this a [try](https://github.com/NixOS/nixpkgs/pull/289187). Still very much WIP, but it's working so far for my current project.

**Comment by dwf** [comment]
@Madouura First, thanks for all your work on this front.

You left a comment to the effect that rocBLASLt is "Very broken with Tensile at the moment, only supports GFX9". It _looks_ like other platforms might be supported now, but I wondered if you might be able to elaborate with the "very broken with Tensile" part.  I notice that they ship a vendored "Tensilelite", was that what you were trying to use?

Any pointers you have on how I might manage to build this would be useful. I'm currently eyeing the rocBLAS derivation as a potentially good starting point.

**Comment by SomeoneSerge** [comment]
I edited the description to add an entry for rocblaslt. It's, apparently, a dependency for zluda

**Comment by jalil-salame** [comment]
Apparently pytorch now requires `hipBLASLt`:

```
python3.11-torch> CMake Error at cmake/public/LoadHIP.cmake:37 (find_package):
python3.11-torch>   By not providing "Findhipblaslt.cmake" in CMAKE_MODULE_PATH this project
python3.11-torch>   has asked CMake to find a package configuration file provided by
python3.11-torch>   "hipblaslt", but CMake did not find one.
python3.11-torch>   Could not find a package configuration file provided by "hipblaslt" with
python3.11-torch>   any of the following names:
python3.11-torch>     hipblasltConfig.cmake
python3.11-torch>     hipblaslt-config.cmake
python3.11-torch>   Add the installation prefix of "hipblaslt" to CMAKE_PREFIX_PATH or set
python3.11-torch>   "hipblaslt_DIR" to a directory containing one of the above files.  If
python3.11-torch>   "hipblaslt" provides a separate development package or SDK, be sure it has
python3.11-torch>   been installed.
python3.11-torch> Call Stack (most recent call first):
python3.11-torch>   cmake/public/LoadHIP.cmake:160 (find_package_and_print_version)
python3.11-torch>   cmake/Dependencies.cmake:1258 (include)
python3.11-torch>   CMakeLists.txt:754 (include)
python3.11-torch>
python3.11-torch> -- Configuring incomplete, errors occurred!
```

**Comment by ony** [comment]
As per https://github.com/pytorch/pytorch/issues/119081#issuecomment-2166504992 in 2.4.0+ (future release) it should be possible to use something like:

```nix
  pythonPackagesExtensions = prev.pythonPackagesExtensions ++ [
    (python-final: python-prev: {
      torch = python-prev.torch.overrideDerivation (oldAttrs: {
        TORCH_BLAS_PREFER_HIPBLASLT = 0;  # not yet in nixpkgs
      });
    })
  ];
```

**Comment by AngryLoki** [comment]
@ony , TORCH_BLAS_PREFER_HIPBLASLT is environment variable for runtime; pytorch still links and requires hipblaslt, even when unused. https://github.com/pytorch/pytorch/pull/120551 should help, but I have no idea whether and when it could be accepted.

By the way, hipblaslt is not difficult to build. Just don't build 6.0 release, skip directly to 6.1. When I tried, bundled TensileLine in 6.0 generated wall of unreadable errors, while 6.1 worked from first attempt.

**Comment by DerDennisOP** [comment]
I'm not able to build rocmlir-rock-6.0.2, when trying to install zluda.

```
FAILED: mlir/lib/Dialect/Rock/Transforms/CMakeFiles/obj.MLIRRockTransforms.dir/ViewToTransform.cpp.o
/nix/store/16pvlpl13g06f1rqxp7z0il9i4l9mlww-rocm-llvm-clang-wrapper-6.0.2/bin/clang++ -DGTEST_HAS_RTTI=0 -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/
build/source/build/mlir/lib/Dialect/Rock/Transforms -I/build/source/mlir/lib/Dialect/Rock/Transforms -I/build/source/external/llvm-project/llvm/include -I/build/source/build/external/llvm-project/llvm/include -I/build/source/external/llv
m-project/mlir/include -I/build/source/build/external/llvm-project/llvm/tools/mlir/include -I/build/source/external/mlir-hal/mlir/include -I/build/source/build/external/mlir-hal/include -I/build/source/external/mlir-hal/include -I/build/
source/mlir/include -I/build/source/build/mlir/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wm
issing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmislead
ing-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused
-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsugg
est-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Werror=global-constructors -O3 -DNDEBUG -std=gnu++17 -fPIC   -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LI
BCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exception
s -funwind-tables -fno-rtti -UNDEBUG -MD -MT mlir/lib/Dialect/Rock/Transforms/CMakeFiles/obj.MLIRRockTransforms.dir/ViewToTransform.cpp.o -MF mlir/lib/Dialect/Rock/Transforms/CMakeFiles/obj.MLIRRockTransforms.dir/ViewToTransform.cpp.o.d
-o mlir/lib/Dialect/Rock/Transforms/CMakeFiles/obj.MLIRRockTransforms.dir/ViewToTransform.cpp.o -c /build/source/mlir/lib/Dialect/Rock/Transforms/ViewToTransform.cpp
In file included from /build/source/mlir/lib/Dialect/Rock/Transforms/ViewToTransform.cpp:14:
/build/source/mlir/include/mlir/Conversion/TosaToRock/TosaToRock.h:21:10: fatal error: 'mlir/Conversion/RocMLIRPasses.h.inc' file not found
#include "mlir/Conversion/RocMLIRPasses.h.inc"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Is there an easy fix for it?

**Comment by AngryLoki** [comment]
@DerDennisOP , it was addressed in pull-request https://github.com/ROCm/rocMLIR/pull/1640 (issue https://github.com/ROCm/rocMLIR/issues/1620), you may want use it.

**Comment by ilylily** [comment]
@DerDennisOP @AngryLoki i think you'll actually also need ROCm/rocMLIR#1542 (closes ROCm/rocMLIR#1500). similar patch in a nearby file

**Comment by arunoruto** [comment]
> @DerDennisOP , it was addressed in pull-request [ROCm/rocMLIR#1640](https://github.com/ROCm/rocMLIR/pull/1640) (issue [ROCm/rocMLIR#1620](https://github.com/ROCm/rocMLIR/issues/1620)), you may want use it.


> @DerDennisOP @AngryLoki i think you'll actually also need [ROCm/rocMLIR#1542](https://github.com/ROCm/rocMLIR/pull/1542) (closes [ROCm/rocMLIR#1500](https://github.com/ROCm/rocMLIR/issues/1500)). similar patch in a nearby file

Is there a plan to patch these things in upstream too? As far as I can see, the [hydro logs](https://hydra.nixos.org/build/278281553/nixlog/1/tail) show the same error as @DerDennisOP.
Would a general update to a newer version resolve this problem? Maybe updating everything to `6.2.4` or even `6.3.0` would be feasible.

**Comment by mschwaig** [comment]
Right now I do not have the time to update ROCm, but I could help out as a reviewer.
I think updating to a newer version would be a great idea.

**Comment by arunoruto** [comment]
> Right now I do not have the time to update ROCm, but I could help out as a reviewer. I think updating to a newer version would be a great idea.

While I do not have that much experience with ROCm, I could try it.
I was also wondering, there are a lot of PRs from the autoupdater @r-ryantm, for example #358391. If the nixpkgs-review succeeds on these PRs, would they be directly viable for a marge? Do we need to bring all the packages up to date at the same time? Are there more things to consider?

**Comment by bgamari** [comment]
> Would a general update to a newer version resolve this problem? Maybe updating everything to `6.2.4` or even `6.3.0` would be feasible.

FWIW, I have a [branch](https://github.com/bgamari/nixpkgs/tree/wip/rocm/rocm-6.2.4) where I have tried updating things to 6.2.4. Unfortunately, I am seeing testsuite failures in the `clang` (specifically `libcxx`) bootstrap that I doubt I will have time to fix in the near-term.

**Comment by bgamari** [comment]
FWIW, I have opened a draft MR to record the state of my attempt: https://github.com/NixOS/nixpkgs/pull/364423

**Comment by LunNova** [comment]
It may now be infeasible for hydra to build some critical rocm packages on its current runners in a 10 hour time limit.
Some packages take hundreds of GB of RAM or hundreds of GB of /build even with a restricted target set.
composable_kernel for 6.3.0 being built for just two gfx targets with a current gen 128c build server takes ~3 hours and peaks over 200GB of RAM. Increasing this to the full target list + running on a less powerful builder will not complete within the limit.

**Comment by LunNova** [comment]
I think it's necessary to design a scheme that allows `-minimal` versions of the largest rocm packages to be built with support for no targets built in and then use env vars which direct the libraries to load code objects for the current hardware from target specific packages, instead of requiring mega packages built for all targets or rebuilding an entire package set for your specific targets.

**Comment by AngryLoki** [comment]
@LunNova , regarding memory/time consumption you may want to disable `-amdgpu-early-inline-all=true` in [composable-kernel](https://github.com/gentoo/gentoo/blob/582df03a0d0cb5e61661f64ee629e8d2d0e9ba6b/sci-libs/composable-kernel/files/composable-kernel-6.3.0-no-inline-all.patch) and [hipcc](https://github.com/gentoo/gentoo/blob/582df03a0d0cb5e61661f64ee629e8d2d0e9ba6b/dev-util/hipcc/hipcc-6.3.0.ebuild#L73-L75), the compilation time is unbearable for mortals. See https://github.com/llvm/llvm-project/issues/86332 for the rationale.
Overall I'm searching for solution for this question too. LLVM 19 added new targets like [gfx9-generic](https://llvm.org/docs/AMDGPUUsage.html#generic-processor-versioning:~:text=Target%20Restrictions-,gfx9%2Dgeneric,-amdgcn) so there are 6 targets at most, but even then it is still too much.

**Comment by LunNova** [comment]
In the future amdgcnspirv might be a solution: https://github.com/ROCm/ROCm/issues/3985#issuecomment-2552432623

**Comment by LunNova** [comment]
I'm not seeing a significant improvement in composable_kernel compilation time from applying https://github.com/gentoo/gentoo/blob/582df03a0d0cb5e61661f64ee629e8d2d0e9ba6b/sci-libs/composable-kernel/files/composable-kernel-6.3.0-no-inline-all.patch

This is on rocm-6.3.0 + a few patches, some related to the inline pass performance issues. https://github.com/LunNova/llvm-project-rocm/commits/lunnova/rocm-6.3.x-optimized/

**Comment by Snektron** [comment]
> Overall I'm searching for solution for this question too. LLVM 19 added new targets like [gfx9-generic](https://llvm.org/docs/AMDGPUUsage.html#generic-processor-versioning:~:text=Target%20Restrictions-,gfx9%2Dgeneric,-amdgcn) so there are 6 targets at most, but even then it is still too much.

Be advised that these targets are often not optimized for. For example, rocPRIM does not have tunings for these architectures.

**Comment by Shawn8901** [comment]
fyi: rocm toolchain seems to be mostly broken on staging-next ( https://hydra.nixos.org/eval/1810617?filter=rocm&compare=1810613&full=#tabs-still-fail ). Not sure if someone here might want to take a look / has an idea.
might be worth to take a look before staging-next get merged. Test builds should be quite easy as most of the packages are already on the cache.

**Comment by AngryLoki** [comment]
> I'm not seeing a significant improvement in composable_kernel compilation

You are using `-DCMAKE_CXX_COMPILER=hipcc` when building `composable_kernel`, and hipcc adds an extra hidden `-mllvm -amdgpu-early-inline-all=true`, that's why `https://github.com/gentoo/gentoo/blob/582df03a0d0cb5e61661f64ee629e8d2d0e9ba6b/dev-util/hipcc/hipcc-6.3.0.ebuild#L73-L75` removes it too. The effect of this flag is **very** noticeable.

As alternative solution, `CMAKE_CXX_COMPILER=clang++` can compile hip code now pretty well without hidden flags (but this is novel).

**Comment by LunNova** [comment]
> You are using `-DCMAKE_CXX_COMPILER=hipcc` when building `composable_kernel`

```
$ nix log .#rocmPackages.composable_kernel_build | grep CMAKE_CXX_COMPILER                                            
CMAKE_CXX_COMPILER: /nix/store/45aqs6snwhnnhsvpbjrbrf4hfsk2pdix-clr-6.3.1/bin/clang++
```

**Comment by AngryLoki** [comment]
> > You are using `-DCMAKE_CXX_COMPILER=hipcc` when building `composable_kernel`
> 
> ```
> $ nix log .#rocmPackages.composable_kernel_build | grep CMAKE_CXX_COMPILER                                            
> CMAKE_CXX_COMPILER: /nix/store/45aqs6snwhnnhsvpbjrbrf4hfsk2pdix-clr-6.3.1/bin/clang++
> ```

Maybe it is overriden, I checked in https://hydra.nixos.org/build/253114834/log - `-DCMAKE_CXX_COMPILER=clang++ ... -DCMAKE_CXX_COMPILER=hipcc` set in https://github.com/NixOS/nixpkgs/blob/7ba25148dc89037ded10338d126e233758679156/pkgs/development/rocm-modules/6/composable_kernel/default.nix#L55

**Comment by LunNova** [comment]
Have been testing in this repo: https://github.com/LunNova/ml.nix/blob/main/rocm-6/composable_kernel/default.nix

**Comment by LunNova** [comment]
Turned the out of tree 6.3 into a draft PR: https://github.com/NixOS/nixpkgs/pull/367695

**Comment by LunNova** [comment]
Related issue for packaging the ROCm out-of-tree amdgpu driver: https://github.com/NixOS/nixpkgs/issues/366242

Not needed for single GPU setups, required for infiniband/ROCe clusters, probably required for multi-GPU single node PCIE P2P to work (DMABUF IPC support for mainline appears to be broken).

**Comment by yshui** [comment]
hi, can we have `gpuTargets` be globally overridable? currently i have to override the rocm packages one-by-one.

you can do that with `cudaPackages` by changing `cudaFlags.cudaCapabilities`

**Comment by mschwaig** [comment]
I opened an issue to remove `rocmPackages_5`: https://github.com/NixOS/nixpkgs/pull/378226

**Comment by LunNova** [comment]
@yshui Approach in https://github.com/NixOS/nixpkgs/issues/366242 includes support for overriding targets for the package set

**Comment by yshui** [comment]
@LunNova nice! looking forward to it.

**Comment by DerDennisOP** [comment]
Build failure rocmlir-rock:
https://hydra.nixos.org/build/289929925/log

**Comment by mschwaig** [comment]
I have opened an issue about updating the list of ROCm maintainers: https://github.com/NixOS/nixpkgs/issues/385288

The point is to add active contributors and remove inactive ones.

**Comment by mschwaig** [comment]
I have also opened an issue about disabling the @r-ryantm bot for ROCm: https://github.com/NixOS/nixpkgs/issues/385294

**Comment by mweinelt** [comment]
For migraphx there are python bindings that would be useful for ROCm support in [Frigate](https://frigate.video/).

This is the code that would need to be built: https://github.com/ROCm/AMDMIGraphX/blob/develop/src/py/migraphx_py.cpp

**Comment by fangpenlin** [comment]
Hi there, not sure why rocm_bandwidth_test marked as "not needed", but I actually need it while trying to fix ROCm peer-to-peer PCI communication issues, so it's pretty useful tool. So I added the package for myself and also open a PR here:

https://github.com/NixOS/nixpkgs/pull/415742

**Comment by mweinelt** [comment]
The rocm detector is unfortunately going away in 0.16.0.

https://github.com/blakeblackshear/frigate/pull/16913

The way forward seems to be through onnx.

**Comment by GZGavinZhao** [comment]
@kamo104 Have you tried setting the environment variable `HSA_OVERRIDE_GFX_VERSION=9.0.0`?

**Comment by kamo104** [comment]
@GZGavinZhao 
Yes I did, but it resulted in the same error message. I tried 10.0.0-3 as well. Also, frigate automatically sets this variable as specified in [the doc](https://github.com/blakeblackshear/frigate/blob/898d1de8750342df00cd2df0ba263249e9405bb3/docs/docs/configuration/object_detectors.md?plain=1#L443).

**Comment by Stzx** [comment]
`libmigraphx_gpu.so` looks broken

```
> /nix/store/wbsh9bn1sqj1jkpc2gcwxpdps07mqy5a-migraphx-6.3.3/bin/migraphx-driver --help

/nix/store/wbsh9bn1sqj1jkpc2gcwxpdps07mqy5a-migraphx-6.3.3/bin/migraphx-driver: symbol lookup error: /nix/store/wbsh9bn1sqj1jkpc2gcwxpdps07mqy5a-migraphx-6.3.3/bin/../lib/migraphx/lib/libmigraphx_gpu.so.2011000: undefined symbol: _ZN8migraphx14version_2_11_03gpu17is_module_fusibleERKNS0_6moduleERKNS1_7contextERKNS0_5valueE
```

**Comment by expenses** [comment]
Would it be possible to add 1103 to gpuTargets? rocmblas doesn't work on my system despite upstream support: https://github.com/ROCm/rocm-libraries/pull/1320

```
rocBLAS error: Cannot read /nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary.dat: Illegal seek for GPU arch : gfx1103
 List of available TensileLibrary Files :
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1010.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1030.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1100.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1101.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1102.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1150.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1151.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1200.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx1201.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx900.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx906.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx908.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx90a.dat"
"/nix/store/xmmzi8raw2ylfvgm4c1lkbzldysgly5a-rocblas-7.2.2/lib/rocblas/library/TensileLibrary_lazy_gfx942.dat"
```

**Comment by LunNova** [comment]
> Would it be possible to add 1103 to gpuTargets? rocmblas doesn't work on my system despite upstream support: [ROCm/rocm-libraries#1320](https://github.com/ROCm/rocm-libraries/pull/1320)

Probably fine to add, will give it a shot.
